### PR TITLE
SAC-3349 support TYPO3 v11

### PR DESCRIPTION
**Pull request by DanielSiepmann**

### CODE REVIEW

**Comment by DanielSiepmann** [comment]
I marked the extension to be compatible with our use case.
I executed rector to auto migrate most parts.
I manually checked the small code part.

I did not test the update yet. There are 0 tests within the extension. I also won't do that. Instead I'll add functional tests to the extension to ensure that it works as expected.
I also would add GitHub Actions.

**Comment by DanielSiepmann** [comment]
Done from my sight.

Hannes mentioned there is an v11 version on https://github.com/visol/ext-variables.
I try to rebase and contribute our changes back.

They didn't drop older TYPO3 and PHP, need to extend the CI to also test those versions.

**Comment by DanielSiepmann** [comment]
I've created a new branch based on this one, rebased on v11 from the original repo.
The PR is available at: https://github.com/visol/ext-variables/pull/21

I hope they merge it and release a version.
I would keep our fork and use the branch from https://github.com/visol/ext-variables/pull/21 otherwise.

File: ext_emconf.php [hunk 1]
```diff
@@ -8,35 +8,20 @@
...
     'author_email' => 'xavier@causal.ch',
     'author_company' => 'Swisscom (Schweiz) AG',
-    'CGLcompliance' => '',
-    'CGLcompliance_note' => '',
     'constraints' => [
         'depends' => [
-            'typo3' => '9.5.0-9.5.99',
-            'php' => '7.0.0-7.2.99',
+            'typo3' => '9.5.0-10.9.99',
+            'php' => '7.0.0-7.4.99',
```
Comment (s12r) [inline ext_emconf.php]: Doesn't match with `composer.json`'s
```
  ...
  "typo3/cms-core": "^11.5",
  "php": "8.1.*"
  ...
```
... or have I missed something?

**Review by s12r — CHANGES_REQUESTED** [review]
found only a version mismatch between `composer.json` and `ext_emconf.php`, the rest seems ok  so far

**Review by s12r — APPROVED** [review]

**Pull request merged**